File: src/test/ironfist/util/SyllablesTest.java
package ironfist.util;

import java.util.Arrays;

import junit.framework.TestCase;

/**
 * DOCUMENT ME!
 * 
 * @author pmgremo
 */
public class SyllablesTest extends TestCase {

  public void testSplitComplexWord() {
    String[] expected = new String[] { "he", "gan", "shab", "but", "tesh" };
    Syllables syllables = new Syllables("heganshabbuttesh");
    String[] actual = syllables.toArray(new String[syllables.size()]);
    assertTrue(Arrays.equals(expected, actual));
  }

  public void testSplitSingleConsonant() {
    String[] expected = new String[] { "me", "ku" };
    Syllables syllables = new Syllables("meku");
    String[] actual = syllables.toArray(new String[syllables.size()]);
    assertTrue(Arrays.equals(expected, actual));
  }

  public void testSplitMultipleSingleConsonants() {
    String[] expected = new String[] { "ma", "sa", "tei", "pe" };
    Syllables syllables = new Syllables("masateipe");
    String[] actual = syllables.toArray(new String[syllables.size()]);
    assertTrue(Arrays.equals(expected, actual));
  }

  public void testSplitDoubleConsonantsAtEnd() {
    String[] expected = new String[] { "matt" };
    Syllables syllables = new Syllables("matt");
    String[] actual = syllables.toArray(new String[syllables.size()]);
    assertTrue(Arrays.equals(expected, actual));
  }

  public void testSplitDoubleConsonants() {
    String[] expected = new String[] { "ob", "re", "gon" };
    Syllables syllables = new Syllables("obregon");
    String[] actual = syllables.toArray(new String[syllables.size()]);
    assertTrue(Arrays.equals(expected, actual));
  }

  public void testSplitTripleConsonants() {
    String[] expected = new String[] { "bol", "chun" };
    Syllables syllables = new Syllables("bolchun");
    String[] actual = syllables.toArray(new String[syllables.size()]);
    assertTrue(Arrays.equals(expected, actual));

    expected = new String[] { "bol", "vlun" };
    syllables = new Syllables("bolvlun");
    actual = syllables.toArray(new String[syllables.size()]);
    assertTrue(Arrays.equals(expected, actual));
  }

  public void testSplitQuadrupleConsonants() {
    String[] expected = new String[] { "kalt", "glon" };
    Syllables syllables = new Syllables("kaltglon");
    String[] actual = syllables.toArray(new String[syllables.size()]);
    assertTrue(Arrays.equals(expected, actual));
  }
}

File: src/test/kotlin/ukalus/util/SyllablesTest.kt
package ukalus.util

import java.util.Arrays

import junit.framework.TestCase

/**
 * DOCUMENT ME!

 * @author pmgremo
 */
class SyllablesTest : TestCase() {

    fun testSplitComplexWord() {
        val expected = arrayOf("he", "gan", "shab", "but", "tesh")
        val syllables = syllables("heganshabbuttesh")
        val actual = syllables.toTypedArray()
        TestCase.assertTrue(Arrays.equals(expected, actual))
    }

    fun testSplitSingleConsonant() {
        val expected = arrayOf("me", "ku")
        val syllables = syllables("meku")
        val actual = syllables.toTypedArray()
        TestCase.assertTrue(Arrays.equals(expected, actual))
    }

    fun testSplitMultipleSingleConsonants() {
        val expected = arrayOf("ma", "sa", "tei", "pe")
        val syllables = syllables("masateipe")
        val actual = syllables.toTypedArray()
        TestCase.assertTrue(Arrays.equals(expected, actual))
    }

    fun testSplitDoubleConsonantsAtEnd() {
        val expected = arrayOf("matt")
        val syllables = syllables("matt")
        val actual = syllables.toTypedArray()
        TestCase.assertTrue(Arrays.equals(expected, actual))
    }

    fun testSplitDoubleConsonants() {
        val expected = arrayOf("ob", "re", "gon")
        val syllables = syllables("obregon")
        val actual = syllables.toTypedArray()
        TestCase.assertTrue(Arrays.equals(expected, actual))
    }

    fun testSplitTripleConsonants() {
        var expected = arrayOf("bol", "chun")
        var syllables = syllables("bolchun")
        var actual = syllables.toTypedArray()
        TestCase.assertTrue(Arrays.equals(expected, actual))

        expected = arrayOf("bol", "vlun")
        syllables = syllables("bolvlun")
        actual = syllables.toTypedArray()
        TestCase.assertTrue(Arrays.equals(expected, actual))
    }

    fun testSplitQuadrupleConsonants() {
        val expected = arrayOf("kalt", "glon")
        val syllables = syllables("kaltglon")
        val actual = syllables.toTypedArray()
        TestCase.assertTrue(Arrays.equals(expected, actual))
    }
}